packages/astro/src/core/config.ts

```diff
@@ -123,11 +122,18 @@ interface LoadConfigOptions {
 /** Attempt to load an `astro.config.mjs` file */
 export async function loadConfig(options: LoadConfigOptions): Promise<AstroConfig> {
   const root = options.cwd ? path.resolve(options.cwd) : process.cwd();
-  const astroConfigPath = new URL(`./${options.filename || 'astro.config.mjs'}`, `file://${root}/`);
   let userConfig: AstroUserConfig = {};
-  // Load a user-config, if one exists and is provided
-  if (existsSync(astroConfigPath)) {
-    userConfig = (await import(astroConfigPath.href)).default;
+  let userConfigPath: string|undefined;
+
+  if (options.filename) {
+    userConfigPath = options.filename.startsWith('.') ? options.filename : `./${options.filename}`;
+    userConfigPath = fileURLToPath(new URL(userConfigPath, `file://${root}/`))
+  }
+  // Automatically load config file using Proload
+  // If `userConfigPath` is `undefined`, Proload will search for `astro.config.[cm]?[jt]s`
+  const config = await load('astro', { mustExist: false, cwd: root, filePath: userConfigPath });
```

Comment:
This is pretty much all the code. 

If `userConfigPath` is defined, it will load that exact file. Otherwise, it will search for and properly load any of the following files:

```
.
├── astro.config.js
├── astro.config.cjs
├── astro.config.mjs
├── astro.config.ts
├── astro.config.cts
├── astro.config.mts
├── config/
│   ├── astro.config.js
│   ├── astro.config.cjs
│   ├── astro.config.mjs
│   ├── astro.config.ts
│   ├── astro.config.cts
│   └── astro.config.mts
└── package.json#astro
```

Reply:
It also handles a top-level `extends` key which can reference a package preset or a local "base" file for monorepos.

Reply:
```
package.json#astro
```

💯 Yes!